docker/scripts/init

```diff
@@ -147,19 +147,46 @@ cat <<FOREMAN > /tmp/foreman.sh
 #!/bin/bash -e
 source /app/.env
 
+# The database may need to start up for a bit first
+if [ -n "\${INTENTIONALLY_SLEEP}" ]; then
+  echo "Intentionally sleeping \${INTENTIONALLY_SLEEP}"
+  sleep \${INTENTIONALLY_SLEEP}
+fi
+
+if [ -n "\${DATABASE_INITIAL_CONNECT_MAX_RETRIES}" ]; then
+  max=\${DATABASE_INITIAL_CONNECT_MAX_RETRIES}
+  count=0
+  while ! rake database_test:ping > /dev/null 2>&1 && [[ \$count -le \$max ]] ; do
+    count=\$[\$count+1]
+    echo "Retry \$count of \$max attempting to connect to \$DATABASE_HOST. Sleeping \${DATABASE_INITIAL_CONNECT_SLEEP:5}"
+    sleep \${DATABASE_INITIAL_CONNECT_SLEEP:5}
+  done
+fi
+
+# We may need to try and create a database
+if [ -n "\${CREATE_DB}" ]; then
+ sudo -u huginn -EH bundle exec rake db:create
+fi
+
 # Assuming we have a created database, run the migrations and seed it idempotently.
-[ -z "\${DO_NOT_MIGRATE}" ] && sudo -u huginn -EH bundle exec rake db:migrate
-[ -z "\${DO_NOT_SEED}" ] && sudo -u huginn -EH bundle exec rake db:seed
+if [ -z "\${DO_NOT_MIGRATE}" ]; then
+ sudo -u huginn -EH bundle exec rake db:migrate
```

Comment:
I think both would work, but I have never used `db:schema:load` apart from rspec which uses it to create the test database afaik.
Since the migration will once run once per container it probably does not make a huge difference.